packages/docusaurus-theme-classic/src/theme/DocPage/index.js

```diff
@@ -8,17 +8,21 @@
 import React from 'react';
 import {MDXProvider} from '@mdx-js/react';
 
+import siteConfig from '@generated/docusaurus.config';
```

Comment:
Instead of importing siteconfig directly, can you try to consume it from react context instead ? Search for `useDocusaurusContext`

One of the reason is because its more performant on dev due to hot module replacement, and another is module concatenation

Reply:
In fact, I did this initially, but I thought that we could reduce the amount of code in this way.
In the end, we can use nested destructuring, which I done.